src/mbgl/text/collision_feature.cpp

```diff
@@ -82,16 +86,14 @@ void CollisionFeature::bboxifyLabel(const GeometryCoordinates& line, GeometryCoo
     const float firstBoxOffset = -boxSize / 2;
 
     GeometryCoordinate &p = anchorPoint;
-    int index = segment + 1;
+    std::size_t index = segment + 1;
     float anchorDistance = firstBoxOffset;
     const float labelStartDistance = -labelLength / 2;
     const float paddingStartDistance = labelStartDistance - labelLength / 8;
 
     // move backwards along the line to the first segment the label appears on
     do {
-        index--;
-
-        if (index < 0) {
+        if (index == 0u) {
```

Comment:
this whole condition does not seem needed, could be:

```
do {
} while (anchorDistance > paddingStartDistance && --index != 0u)
```

Reply:
I don't think that is correct, `--index != 0u` will short circuit loop early and condition, e.g., `anchorDistance > labelStartDistance)` will never be reachable.